File: rust/contracts/donatewithfeedback/src/schema.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

#![allow(dead_code)]

use wasplib::client::*;
use super::*;

pub const SC_NAME: &str = "donatewithfeedback";
pub const SC_HNAME: ScHname = ScHname(0x696d7f66);

pub const PARAM_AMOUNT: &str = "amount";
pub const PARAM_FEEDBACK: &str = "feedback";

pub const VAR_AMOUNT: &str = "amount";
pub const VAR_DONATIONS: &str = "donations";
pub const VAR_DONATOR: &str = "donator";
pub const VAR_ERROR: &str = "error";
pub const VAR_FEEDBACK: &str = "feedback";
pub const VAR_LOG: &str = "log";
pub const VAR_MAX_DONATION: &str = "maxDonation";
pub const VAR_TIMESTAMP: &str = "timestamp";
pub const VAR_TOTAL_DONATION: &str = "totalDonation";

pub const FUNC_DONATE: &str = "donate";
pub const FUNC_WITHDRAW: &str = "withdraw";
pub const VIEW_DONATIONS: &str = "donations";

pub const HFUNC_DONATE: ScHname = ScHname(0xdc9b133a);
pub const HFUNC_WITHDRAW: ScHname = ScHname(0x9dcc0f41);
pub const HVIEW_DONATIONS: ScHname = ScHname(0x45686a15);

#[no_mangle]
fn on_load() {
    let exports = ScExports::new();
    exports.add_call(FUNC_DONATE, func_donate);
    exports.add_call(FUNC_WITHDRAW, func_withdraw);
    exports.add_view(VIEW_DONATIONS, view_donations);
}

File: contracts/donatewithfeedback/schema.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package donatewithfeedback

import "github.com/iotaledger/wasplib/client"

const ScName = "donatewithfeedback"
const ScHname = client.ScHname(0x696d7f66)

const ParamAmount = client.Key("amount")
const ParamFeedback = client.Key("feedback")

const VarAmount = client.Key("amount")
const VarDonations = client.Key("donations")
const VarDonator = client.Key("donator")
const VarError = client.Key("error")
const VarFeedback = client.Key("feedback")
const VarLog = client.Key("log")
const VarMaxDonation = client.Key("maxDonation")
const VarTimestamp = client.Key("timestamp")
const VarTotalDonation = client.Key("totalDonation")

const FuncDonate = "donate"
const FuncWithdraw = "withdraw"
const ViewDonations = "donations"

const HFuncDonate = client.ScHname(0xdc9b133a)
const HFuncWithdraw = client.ScHname(0x9dcc0f41)
const HViewDonations = client.ScHname(0x45686a15)

func OnLoad() {
    exports := client.NewScExports()
    exports.AddCall(FuncDonate, funcDonate)
    exports.AddCall(FuncWithdraw, funcWithdraw)
    exports.AddView(ViewDonations, viewDonations)
}
